### PR TITLE
improve(across-relayer): Should prune whitelisted L1 tokens using L2 events

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

Currently the user must specify the `RATE_MODELS` for any L2 network that hasn't whitelisted all of the tokens in the default `RATE_MODELS` dictionary in `@uma/sdk`. We should not ask the user to specify `RATE_MODELS` and they should instead be able to rely on the default in the code. This PR prunes the whitelist derived from `RATE_MODELS` for a specific L2 by removing any L1 tokens not on that chain's specific deposit box.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts [hunk 1]
```diff
@@ -62,14 +69,23 @@ export class InsuredBridgeL2Client {
     // Define a config to bound the queries by.
     const blockSearchConfig = {
       fromBlock: this.firstBlockToSearch,
-      toBlock: this.endingBlockNumber || (await this.l2Web3.eth.getBlockNumber()),
```
Comment (chrismaree) [inline packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts:65]: why did you need to make this change? did accessing the `endingBlockNumber` not work? also I don't see why you need to re-declare the `lastBlockToSearch`

File: packages/insured-bridge-relayer/src/RelayerHelpers.ts [hunk 2]
```diff
@@ -35,6 +35,27 @@ export async function approveL1Tokens(
...
+export async function pruneWhitelistedL1Tokens(
+  logger: winston.Logger,
+  l2Client: InsuredBridgeL2Client,
+  whitelistedRelayL1Tokens: string[]
+): Promise<string[]> {
+  await l2Client.update();
+  const filteredWhitelistedRelayL1Tokens = whitelistedRelayL1Tokens.filter((l1TokenAddress: string) => {
+    return l2Client.isWhitelistedToken(l1TokenAddress);
+  });
+  logger.info({
```
Comment (chrismaree) [inline packages/insured-bridge-relayer/src/RelayerHelpers.ts:51]: I dont think this should be an `info` log. this means it'll log to slack on every run

File: packages/insured-bridge-relayer/src/index.ts [hunk 3]
```diff
@@ -52,14 +52,25 @@ export async function run(logger: winston.Logger, l1Web3: Web3): Promise<void> {
...
       config.activatedChainIds[0],
-      l2StartBlock
+      l2StartBlock,
+      l2EndBlock
+    );
+
+    // Update the L2 client and filter out tokens that are not whitelisted on the L2 from the whitelisted
+    // L1 relay list.
+    await l2Client.update();
+    const filteredWhitelistedRelayL1Tokens = await pruneWhitelistedL1Tokens(
```
Comment (chrismaree) [inline packages/insured-bridge-relayer/src/index.ts:70]: if you use a smaller variable name this could fit on one line. how about `filteredL1Whitelist`

File: packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts (hunk 1 above)
Comment (nicholaspai) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts:65]: So i discovered this bug when I passed an `endingBlockNumber` into this function. You'll notice that after each loop, the `fromBlock` gets set to the `toBlock + 1`, but the `toBlock` does not get reset, so after the first loop, the `toBlock` remains at the same value while `fromBlock` is now `> toBlock`.

Comment (nicholaspai) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts:65]: So you'll notice that after each loop, `fromBlock` gets set to `toBlock + 1`, but `toBlock` is not reset. So this causes the next loop to have `fromBlock > toBlock`. Shouldn't we be incrementing `toBlock` as well? Or perhaps we should rename the constructor variable `endingBlockNumber` to `blockSearchWindowLength`, wdyt?

File: packages/insured-bridge-relayer/src/RelayerHelpers.ts (hunk 2 above)
Comment (nicholaspai) [reply packages/insured-bridge-relayer/src/RelayerHelpers.ts]: fixed

File: packages/insured-bridge-relayer/src/index.ts (hunk 3 above)
Comment (nicholaspai) [reply packages/insured-bridge-relayer/src/index.ts]: done

**Review by chrismaree — APPROVED** [review]
